implementations/rust/ockam/ockam_transport_tcp/src/privileged_portal/workers/internal_processor.rs

```diff
@@ -122,15 +110,15 @@ impl Processor for InternalProcessor {
 
                 let portal_packet = OckamPortalPacket::from_tcp_packet(
                     connection.connection_identifier.clone(),
-                    inlet_shared_state.route_index(),
+                    connection.inlet_route_state.route_index(),
```

Comment:
It uses locking internally and locks again to get the route ~10 lines below that. That's a small performance penalty, but also brings a possibility of a data race.

Reply:
Are you right, accessing the data structure with multiple locks and unlocks is bad.
I've created two methods: `snapshot` and `snapshot_route` that return a consistent view.